hadoop-common-project/hadoop-common/src/site/markdown/filesystem/fsdataoutputstreambuilder.md

```diff
@@ -182,3 +182,50 @@ see `FileSystem#create(path, ...)` and `FileSystem#append()`.
     result = FSDataOutputStream
 
 The result is `FSDataOutputStream` to be used to write data to filesystem.
+
+
+## <a name="s3a"></a> S3A-specific options
+
+Here are the custom options which the S3A Connector supports.
+
+| Name                        | Type      | Meaning                                |
+|-----------------------------|-----------|----------------------------------------|
+| `fs.s3a.create.performance` | `boolean` | create a file with maximum performance |
+| `fs.s3a.create.header` | `string` | prefix for user supplied headers |
+
+### `fs.s3a.create.performance`
+
+Prioritize file creation performance over safety checks for filesystem consistency.
+
+This
+1. Skips the `LIST` call which makes sure a file is being created over a directory.
+   Risk: a file is created over a directory.
+1. Ignores the overwrite flag.
+1. Never issues a `DELETE` call to delete parent directory markers.
+   Risk: if directory markers have not already been leaned up, such as when creating the
```

Comment:
typo: `leaned` -> `cleaned`

Reply:
Also from the perspective of a newbie, can you clarify the fallout of the risk? I.e., what happens if I haven't cleaned up the directory marker, and create with the performance flag enabled?

Reply:
things like rename and delete on the file may not find the subdirectory. 
https://github.com/steveloughran/hadoop/blob/s3/HADOOP-17833-magic-committer-performance/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/performance/ITestCreateFileCost.java#L213

nothing is lost, its just that things like treewalking may not find them. This is what that LIST is designed to prevent. Here you say "I know what I am doing" and if you didn't, well, too bad. the price of performance is safety, as BMW and porsche like to point out.